parser/ParseIndexProperties.hpp

```diff
@@ -0,0 +1,412 @@
+          return false;
+        }
+        // All good! allow the default bloom filter size to be modified.
+        index_sub_block_description_->SetExtension(BloomFilterIndexSubBlockDescription::bloom_filter_size,
+                                                   bloom_filter_size);
+      } else if (key.compare(kBloomFilterNumHashes) == 0) {
+        const ParseKeyIntegerValue *key_integer_value = static_cast<const ParseKeyIntegerValue*>(&key_value);
+        if (key_integer_value->value()->float_like()) {
+          setIndexDescriptionAsInvalid(InvalidIndexType::kNumHashesIsFloat, &key_value);
+          return false;
+        }
+        const std::int64_t number_of_hashes = key_integer_value->value()->long_value();
+        if (number_of_hashes < 0) {
+          setIndexDescriptionAsInvalid(InvalidIndexType::kNumHashesIsNegative, &key_value);
+          return false;
+        }
+        // All good! allow the default num_hashes to be modified.
+        index_sub_block_description_->SetExtension(BloomFilterIndexSubBlockDescription::number_of_hashes,
+                                                   number_of_hashes);
+      } else if (key.compare(kBloomFilterProjectElementCount) == 0) {
+        const ParseKeyIntegerValue *key_integer_value = static_cast<const ParseKeyIntegerValue*>(&key_value);
+        if (key_integer_value->value()->float_like()) {
+          setIndexDescriptionAsInvalid(InvalidIndexType::kProjectedCountIsFloat, &key_value);
+          return false;
+        }
+        const std::int64_t projected_element_count = key_integer_value->value()->long_value();
+        if (projected_element_count < 0) {
+          setIndexDescriptionAsInvalid(InvalidIndexType::kProjectedCountIsNegative, &key_value);
+          return false;
+        }
+        // All good! allow the default projected_element_count to be modified.
+        index_sub_block_description_->SetExtension(BloomFilterIndexSubBlockDescription::projected_element_count,
+                                                   projected_element_count);
+      } else {
+        // None of the defined properties matched, this is an invalid property for this index.
+        setIndexDescriptionAsInvalid(InvalidIndexType::kInvalidKey, &key_value);
+        return false;
+      }
+    }
+    invalid_index_type_ = InvalidIndexType::kNone;
+    invalid_property_node_ = nullptr;
+    return true;
+  }
+
+ private:
+  valid_property_map_type valid_property_map_;  // a map of index property constants and their associated type.
+
+  DISALLOW_COPY_AND_ASSIGN(BloomFilterIndexProperties);
+};
+
+
+/**
+ * @brief Implementation of index properties for CSB Tree Index.
+ */
+class CSBTreeIndexProperties : public IndexProperties {
+ public:
+  /**
+   * @brief Constructor.
+   **/
+  CSBTreeIndexProperties()
+    : IndexProperties(new IndexSubBlockDescription(),
+                      InvalidIndexType::kNone,
+                      nullptr) {
```

Comment:
We could remove these two above, given that we have the default values in the `IndexProperties` constructor.